File: runtimecs/java/util/Map.cs
using java.lang;

namespace java.util 
{
    public interface Map
    {
        void clear();
        bool containsKey(object key);
        bool containsValue(object value);
        bool Equals(object o);
        void forEach(java.util.function.BiConsumer biconsumer);
        object get(object key);
        object getOrDefault(object key, object def);
        int GetHashCode();
        bool isEmpty();
        Set keySet();
        object put(object key, object value);
        void putAll(Map m);
        object remove(object key);
        int size();
        Collection values();
        string ToString();
    }
    
    public static class Map_0009
    {
        public static object getOrDefault(Map @this, object key, object def)
        {   
            return @this.containsKey(key) ? @this.get(key) : def;
        }
        public static void forEach(Map @this, java.util.function.BiConsumer biconsumer)
        {
            java.util.Iterator i = @this.keySet().iterator();
            while (i.hasNext()) 
            {   
                object key = i.next();
                object value = @this.get(key);
                biconsumer.accept(key,value);
            }       
        }
    }
}

File: runtimejs/java/util/Map.js
//load// java/lang/Object
var java_util_Map = { 
    _superinterfaces: [],
    _defaults:
    {   getOrDefault_2: function(key,def)
        {   return this.containsKey_1(key) ? this.get_1(key) : def;
        }
        ,
        forEach_1: function(biconsumer)
        {   var i = this.keySet_0().iterator_0();
            while (i.hasNext_0()) 
            {   var k = i.next_0();
                var v = this.get_1(k);
                biconsumer.accept_2(k,v);
            }
        },
    }    
};  

// -- methods:
// void	clear()
// boolean	containsKey(Object key)
// boolean	containsValue(Object value)
// boolean	equals(Object o)
// V	get(Object key)
// int	hashCode()
// boolean	isEmpty()
// Set<K>	keySet()
// V	put(K key, V value)
// void	putAll(Map<? extends K,? extends V> m)
// V	remove(Object key)
// int	size()
// Collection<V>	values()
// String toString()
